robowflex_library/scripts/fetch_visualization.cpp

```diff
@@ -0,0 +1,84 @@
+/* Author: Zachary Kingston */
+/* Modified by: Juan D. Hernandez */
+
+#include <robowflex_library/detail/fetch.h>
+#include <robowflex_library/geometry.h>
+#include <robowflex_library/io/visualization.h>
+#include <robowflex_library/planning.h>
+#include <robowflex_library/robot.h>
+#include <robowflex_library/scene.h>
+#include <robowflex_library/util.h>
+
+using namespace robowflex;
+
+static const std::string GROUP = "arm_with_torso";
+
+int main(int argc, char **argv)
+{
+    // Startup ROS
+    ROS ros(argc, argv);
+
+    // Create the default Fetch robot.
+    auto fetch = std::make_shared<FetchRobot>();
+    fetch->initialize();
+
+    // Dump the geometry information for visualization.
+    fetch->dumpGeometry("fetch.yml");
+
+    // Create an RViz visualization helper. Publishes all topics and parameter under `/robowflex` by
+    // default.
+    IO::RVIZHelper rviz(fetch);
+
+    ROS_INFO("RViz Initialized! Press enter to continue (after your RViz is setup)...");
+    std::cin.get();
+
+    // Create an empty scene.
+    auto scene = std::make_shared<Scene>(fetch);
+    scene->fromYAMLFile("package://robowflex_library/yaml/test_fetch.yml");
+
+    // Visualize the scene.
+    rviz.updateScene(scene);
+
+    // Create the default planner for the UR5.
+    auto planner = std::make_shared<OMPL::FetchOMPLPipelinePlanner>(fetch, "default");
+    planner->initialize();
+
+    // Create a motion planning request with a pose goal.
+    MotionRequestBuilderPtr request(new MotionRequestBuilder(planner, GROUP));
+    fetch->setGroupState(GROUP, {0.05, 1.32, 1.40, -0.2, 1.72, 0.0, 1.66, 0.0});  // Stow
+    request->setStartConfiguration(fetch->getScratchState());
+
+    // Create a motion planning request with a pose goal. Cube3
+    Eigen::Affine3d pose = Eigen::Affine3d::Identity();
+    pose.translate(Eigen::Vector3d{0.4, 0.6, 0.92});
+    Eigen::Quaterniond orn{0.5, -0.5, 0.5, 0.5};
+    auto region = Geometry::makeSphere(0.01);
+    request->setGoalRegion("wrist_roll_link", "world",  // links
+                           pose, region,                // position
+                           orn, {0.1, 0.1, 0.1}         // orientation
+                           );
+
+    ROS_INFO("Scene and Goal displayed! Press enter to plan...");
```

Comment:
The goal is not actually displayed here, missing the call to `rviz.addGoalMarker()` and `rviz.updateMarkers()`